src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,170 @@
+import { useContext, useState, useCallback } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const handleEyeButtonClick = (e) => {
+    e.preventDefault();
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, email: null }));
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, password: null }));
+  };
+
+  const validateEmail = useCallback(() => {
+    const pattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!pattern.test(email)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        email: { valid: false, message: "Не правильна адреса" },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      email: { valid: true },
+    }));
+    return true;
+  }, [email]);
+
+  const validatePassword = useCallback(() => {
+    const pattern = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{9,}$/;
+    if (!pattern.test(password)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        password: {
+          valid: false,
+          message:
+            "Не менше 8 буквенних та 1 числовий символи",
+        },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      password: { valid: true },
+    }));
+    return true;
+  }, [password]);
+
+  const handleSignIn = async (event) => {
+    event.preventDefault();
+    setIsSubmitting(true);
+    if (errors.email?.valid === false || errors.password?.valid === false) {
```

Comment:
1. get rid of `isSubmitting`, use errors instead
2. when checking for errors use programmatic approach, 
do not relay on having only 2 input fields, this may change any time,
also using below approach you "fail-fast", hasErrors will be `true` 
if only first element of the array has error, no need to check all other errors:
  ```js
      const hasErrors = Object.values(errors).some(
        ({ valid }) => valid === false
        //   ^^ 
        //   this is "object distructuring" 
      );

      if (hasErrors) {
        // do whatever you want with errors, 
        // then interrupt the handler with `return`:
        return;
      }
  ```

Reply:
you still have `isSubmitting` state :) 
with this change about you don't needed it anymore.
and remove "disabled" attribute from submit button please - this is bad practice from Accessibility point of view.
here are the explonations:
- https://adrianroselli.com/2024/02/dont-disable-form-controls.html
- https://advancedbytez.com/make-disabled-buttons-accessible